pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -222,14 +221,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 			conditionSynced := NewExternalSecretCondition(esv1beta1.ExternalSecretReady, v1.ConditionTrue, esv1beta1.ConditionReasonSecretDeleted, "secret deleted due to DeletionPolicy")
 			SetExternalSecretCondition(&externalSecret, *conditionSynced)
-			return ctrl.Result{RequeueAfter: requeueAfter}, nil
+			return ctrl.Result{}, err
 
 		case esv1beta1.DeletionPolicyMerge:
 			// noop, handled below
 
 		// In case provider secrets don't exist the kubernetes secret will be kept as-is.
 		case esv1beta1.DeletionPolicyRetain:
-			return ctrl.Result{RequeueAfter: requeueAfter}, nil
+			return ctrl.Result{}, err
```

Comment:
Same here. Even if the data is nil, the error is also nil, which would cause a never sync on the reconcile.

Reply:
good catch, fixed!